### PR TITLE
Update to frontend components 3.x and webpack 5

### PR DESCRIPTION
**Pull request by lgalis**
Update to frontend components 3.x and webpack 5

https://github.com/RedHatInsights/cloud-services-config/pull/458

### CODE REVIEW

**Comment by Hyperkid123** [comment]
@lgalis I had to do some more deps cleaning to get this building properly. After following patch and removing `package-lock.json` and re-installing it started to work for me again:

```diff
diff --git a/package.json b/package.json
index 2ca5b29..9b6227a 100644
--- a/package.json
+++ b/package.json
@@ -66,20 +66,16 @@
     "babel-core": "7.0.0-bridge.0",
     "babel-eslint": "^10.0.3",
     "babel-jest": "^25.0.0",
-    "babel-loader": "^8.0.0-beta.6",
     "babel-plugin-lodash": "^3.3.4",
     "babel-plugin-transform-imports": "^2.0.0",
-    "css-loader": "^5.1.3",
     "enzyme": "^3.11.0",
     "enzyme-adapter-react-16": "^1.15.5",
     "enzyme-to-json": "^3.6.1",
     "eslint": "^7.22.0",
     "eslint-config-prettier": "^8.1.0",
-    "eslint-loader": "^4.0.2",
     "eslint-plugin-formatjs": "^2.12.8",
     "eslint-plugin-prettier": "^3.3.1",
     "eslint-plugin-react": "^7.22.0",
-    "file-loader": "^4.3.0",
     "glob": "^7.1.6",
     "identity-obj-proxy": "^3.0.0",
     "jest": "^25.1.0",
@@ -88,11 +84,7 @@
     "npm-run-all": "^4.1.5",
     "prettier": "^2.2.1",
     "redux-mock-store": "^1.5.4",
-    "sass-loader": "^11.0.1",
-    "source-map-loader": "^2.0.1",
-    "style-loader": "^2.0.0",
     "webpack": "^5.27.2",
-    "webpack-bundle-analyzer": "^4.4.0",
     "webpack-dev-server": "^3.11.2",
     "xhr-mock": "^2.5.1"
   },

```

File: src/App.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ const App = () => {
       .getUser()
       .then(() =>
         getRbacRoleApi()
-        .listRoles(defaultSettings.limit, 0, 'Approval ', 'partial', 'principal')
+        .listRoles(defaultSettings.limit, 0, 'Approval ', '', 'partial', 'principal')
```
Comment (Hyperkid123) [inline src/App.js:38]: I assume there was a change in the client arguments so you have to include the empty string?

File: src/AppEntry.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { BrowserRouter as Router } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import store from './utilities/store';
+import App from './App';
+import { getBaseName } from '@redhat-cloud-services/frontend-components-utilities/helpers/helpers';
```
Comment (Hyperkid123) [inline src/AppEntry.js:6]: We should not use second level import paths, that breaks automatic cjs/ems module resolution

```diff
-import { getBaseName } from '@redhat-cloud-services/frontend-components-utilities/helpers/helpers';
+import { getBaseName } from '@redhat-cloud-services/frontend-components-utilities/helpers';
```

File: src/index.html [hunk 3]
```diff
@@ -2,7 +2,7 @@
 <html lang="en-US">
     <head>
         <meta charset="UTF-8">
-        <title>Approval</title>
+        <title>Approval test</title>
```
Comment (Hyperkid123) [inline src/index.html:5]: I think the template should not include `test`.

**Review by Hyperkid123 — COMMENTED** [review]
Overall looks the migration looks good. I have just some small nitpicks.

File: src/App.js (hunk 1 above)
Comment (lgalis) [reply src/App.js:38]: Yes, a new parameter was added to this endpoint, so this new param is needed

**Review by Hyperkid123 — COMMENTED** [review]
Looks good. Feel free to merge whenever you feel like it. One question though did you try running the app in ci stable? We want to check if it works in older chrome as well. (I have to much stuff to do right now to check it)

**Comment by lgalis** [comment]
@Hyperkid123 - thank you. Will try in ci stable and let you know.

**Comment by lgalis** [comment]
@Hyperkid123 - works with the prod chrome with the store param change ( last commit )

**Review by Hyperkid123 — APPROVED** [review]
Looks good, only the linter has failed.

**Comment by lgalis** [comment]
@Hyperkid123 - remove redux logger for now - will add it back later

**Review by Hyperkid123 — APPROVED** [review]
Feel free to merge whenever you need. :+1:

**Pull request merged**